worker/manager/src/mirrors_qa_manager/docker.py

```diff
@@ -0,0 +1,141 @@
+                        logger.error(
+                            f"docker api error for {func.__name__} "
+                            f"(attempt {attempt}): {exc}"
+                        )
+                        time.sleep(interval * attempt)
+                        continue
+                    raise exc
+
+        return wrapped
+
+    return decorator
+
+
+def get_running_container_name() -> str:
+    """Determine the name of this container."""
+    with open("/etc/hostname") as fh:
+        return Path(fh.read().strip()).name
+
+
+def get_container(client: DockerClient, container_name: str) -> Container:
+    return client.containers.get(
+        container_name
+    )  # pyright: ignore [reportGeneralTypeIssues,reportReturnType]
+
+
+@retry()
+def get_docker_client() -> DockerClient:
+    logger.info(f"Connecting to Docker API on {Settings.DOCKER_SOCKET}")
+    client = DockerClient(
+        base_url=f"unix://{Settings.DOCKER_SOCKET}",
+        timeout=Settings.DOCKER_CLIENT_TIMEOUT,
+    )
+    client.ping()
+    logger.info("Connected to Docker API successfully")
+    return client
+
+
+@retry()
+def remove_container(
+    client: DockerClient,
+    container_id: str,
+    *,
+    remove_volumes: bool = False,
+    force: bool = False,
+    not_found_ok: bool = False,
+) -> None:
+    try:
+        container = get_container(client, container_id)
+    except NotFound as exc:
+        if not_found_ok:
+            pass
+        else:
+            raise exc
+    else:
+        container.remove(  # pyright: ignore [reportGeneralTypeIssues]
+            v=remove_volumes, force=force
+        )
+
+
+@retry()
+def get_or_pull_image(client: DockerClient, name: str) -> Image:
+    logger.debug(f"Getting image {name}")
+    # task worker is always pulled to ensure we can update our code
+    if ":" not in name:
+        # consider missing :tag info as a local image for tests
+        return client.images.get(
+            name
+        )  # pyright: ignore[reportGeneralTypeIssues,reportReturnType]
+
+    return client.images.pull(
+        name
+    )  # pyright: ignore[reportGeneralTypeIssues,reportReturnType]
+
+
+def run_container(client: DockerClient, image_name: str, **kwargs: Any) -> Container:
+    image = get_or_pull_image(client, image_name)
+    return client.containers.run(
+        image, **kwargs
+    )  # pyright: ignore[reportGeneralTypeIssues, reportReturnType]
+
+
+def query_host_mounts(client: DockerClient, keys: list[Path]) -> dict[Path, Path]:
+    return query_container_mounts(
+        get_container(client, get_running_container_name()), keys
+    )
+
+
+def query_container_mounts(container: Container, keys: list[Path]) -> dict[Path, Path]:
+    mounts = {}
+    for mount in container.attrs["Mounts"]:
+        dest = Path(mount["Destination"])
+        if dest in keys:
+            key = keys[keys.index(dest)]
+            mounts[key] = Path(mount["Source"])
+    return mounts
+
+
+@retry()
```

Comment:
```suggestion
@retry
```

Reply:
Unresolving this because of the comments further down. Should I switch back to the explicit style?